docs/single_entrypoint_proxy_contract.md

```diff
@@ -0,0 +1,383 @@
+    "deadline": 1000000
+  },
+  "value": {
+    "name": "Alan Turing",
+    "place": "England"
+  }
+}
+```
+One thing to note is when these values are present outside of the context of the OpenWhisk control plane, they may not actually be used for anything. However, the `activation` key is still necessary to signal the intent to run the function.
+
+## Passing parameters
+
+Similar to the description of the `value` key in the `activation` object during the [activation of standard OpenWhisk actions](actions-new.md#activation), a top level `value` key in the JSON object passed to the `/` endpoint (with a corresponding top level `activation` key) is how parameters are passed to the underlying function being run.
+In the following example:
+```json
+{
+  "activation": {
+    "namespace": "",
+    "action_name": "hello",
+    "api_host": "",
+    "api_key": "",
+    "activation_id": "",
+    "transaction_id": "",
+    "deadline": 1000000
+  },
+  "value": {
+    "name": "Alan Turing",
+    "location": "England"
+  }
+}
+```
+
+The underlying function would recieve a parameters map with the keys `name` and `location` with the values `Alan Turing` and `England` respectively.
+
+## Init/Run
+
+OpenWhisk stem cell runtimes being executed in a single entrypoint execution environment can be both initialized and activated at the same time by passing both `init` and `activation` keys in the same JSON object to the `/` endpoint. This will first initialize the runtime, following the same procedures described above, and then subsequently activate the same runtime.
+For example:
+```json
+{
+  "init": {
+    "name" : "hello",
+    "main" : "main",
+    "code" : "function main(params) {return { payload: 'Hello ' + params.name + ' from ' + params.place +  '!' };}",
+    "binary": false,
+    "env": {}
+  },
+  "activation": {
+    "namespace": "",
+    "action_name": "hello",
+    "api_host": "",
+    "api_key": "",
+    "activation_id": "",
+    "transaction_id": "",
+    "deadline": 1000000
+  },
+  "value": {
+    "name": "Alan Turing",
+    "location": "England"
+  }
+}
+```
+The above JSON object would instruct the runtime to be initialized with the function under `init.code` and be run with the function being passed the object `{name: "Alan Turing", location: "England"}`. It would then return the JSON object
+```json
+{
+  "payload": "Hello Alan Turing from England!"
+}
+```
+
+## Example Cases
+Below is a table outlining the standardized behaviors that any action proxy implementation needs to fulfill. NodeJS was the sample language used, but corresponding example cases could be written in the language of the corresponding runtime it is showcasing.
+
+
+<table border="2" cellspacing="0" cellpadding="6" rules="groups" frame="hsides">
+
+
+<colgroup>
+<col/>
+
+<col/>
+
+<col/>
+
+<col/>
+
+<col/>
+
+<col/>
+
+<col/>
+</colgroup>
+<thead>
+<tr>
+<th scope="col" >Test Name</th>
+<th scope="col" >Action Code (In NodeJS)</th>
+<th scope="col" >Input</th>
```

Comment:
Can we please link to the JSON files since they are now merged into openwhisk-test?

Reply:
They are linked further down in the Nodejs table